### PR TITLE
Fix cross compile issue introduced in 5.16

### LINKED ISSUE #9951: cross build regressed wrt System.Native and "shm_open that works well enough with mmap"
## Steps to Reproduce

1. on e.g. linux amd64: 
`./autogen.sh  CFLAGS="-g -O0" -disable-nls -disable-system-aot --host=arm-linux-gnueabihf CC=arm-linux-gnueabihf-gcc -enable-cross-compiling`

### Current Behavior

```
.
.
.
checking for shm_open that works well enough with mmap... configure: error: in `/s/mono3':
configure: error: cannot run test program while cross compiling
See `config.log' for more details
```

### Expected Behavior

Uncertain. Perhaps success and compile and link, perhaps not.
Maybe cross builds too esoteric.

We have defaults in some places and don't run programs.

```
      dnl ***********************************
        dnl *** Checks for working __thread ***
        dnl ***********************************
        AC_MSG_CHECKING(for working __thread)
        if test "x$with_tls" != "x__thread"; then
                AC_MSG_RESULT(disabled)
        elif test "x$cross_compiling" = "xyes"; then
                AC_MSG_RESULT(cross compiling, assuming yes)
        else
                AC_TRY_RUN([
  
```

and

```
        dnl **************************************
        dnl *** Checks for working sigaltstack ***
        dnl **************************************
        AC_MSG_CHECKING(for working sigaltstack)
        if test "x$with_sigaltstack" != "xyes"; then
                AC_MSG_RESULT(disabled)
        elif test "x$cross_compiling" = "xyes"; then
                AC_MSG_RESULT(cross compiling, assuming yes)
        else
                AC_TRY_RUN([
```

but:

```
       AC_MSG_CHECKING(for shm_open that works well enough with mmap)
        if test "x$ac_cv_func_shm_open" = "xno" -o "x$ac_cv_func_shm_open_working_with_mmap" = "xno" ; then
                AC_MSG_RESULT(no)
        else
                AC_TRY_RUN([
 ```

therefore:

```
checking for shm_open that works well enough with mmap... configure: error: in `/s/mono3':
configure: error: cannot run test program while cross compiling
```

Is there a good default maybe?
This came in prematurely with System.Native ahead of its memory mapped file support?
Which is coming?

Anyway, I'll just set ac_cv_func_shm_open_working_with_mmap and move on.

## On which platforms did you notice this

Linux

### PR DESCRIPTION
**Pull request by ShonkyCH**
Fixes #9951

### CODE REVIEW

**Comment by marek-safar** [comment]
approve

File: configure.ac [hunk 1]
```diff
@@ -3299,6 +3299,8 @@ if test x$host_win32 = xno; then
 	AC_MSG_CHECKING(for shm_open that works well enough with mmap)
 	if test "x$ac_cv_func_shm_open" = "xno" -o "x$ac_cv_func_shm_open_working_with_mmap" = "xno" ; then
 		AC_MSG_RESULT(no)
+	elif test "x$cross_compiling" = "xyes"; then
+		AC_MSG_RESULT(cross compiling, assuming yes)
```
Comment (alexischr) [inline configure.ac]: if assuming yes, this block also needs: 
`AC_DEFINE(HAVE_SHM_OPEN_THAT_WORKS_WELL_ENOUGH_WITH_MMAP, 1, [shm_open that works well enough with mmap])`

Comment (alexischr) [reply configure.ac]: The `sigaltstack` check was also not fully doing the right thing so thank you for bringing attention to this: https://github.com/mono/mono/pull/12365

**Comment by luhenry** [comment]
@alexischr could you please update the PR?

**Comment by alexischr** [comment]
@monojenkins squash

**Comment by lewurm** [comment]
@monojenkins build failed

**Comment by alexischr** [comment]
@monojenkins backport to 2018-12

**Comment by alexischr** [comment]
@monojenkins backport to 2018-10

**Comment by alexischr** [comment]
@monojenkins backport to 2018-08

**Comment by alexischr** [comment]
@monojenkins backport to 2018-06

**Pull request merged**